apps/web/src/hooks/useDebounce.ts

```diff
@@ -1,13 +1,15 @@
-import { useCallback, useEffect, useRef } from 'react';
+import { useCallback, useEffect } from 'react';
 import debounce from 'lodash.debounce';
 
+import { useDataRef } from './useDataRef';
+
 export const useDebounce = <Arguments = unknown | unknown[]>(callback: (args: Arguments) => void, ms = 0) => {
-  const callbackRef = useRef(callback);
-  callbackRef.current = callback;
+  const callbackRef = useDataRef(callback);
 
-  const debouncedCallback = useCallback(debounce(callbackRef.current, ms), [ms]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedCallback = useCallback(debounce(callbackRef.current, ms), [callbackRef, ms]);
```

Comment:
unfortunately, I had to use this here, because we don't have the types for the `debounce` and it's not a simple thing, it's basically the function that could be called but it has also a cancel method, so it's not also easy to type is by hands...